### PR TITLE
ET-4985 add tests

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## What 🕵️ 🔍

- add api tests
- [x] add tenant header https://github.com/xaynetwork/xayn_infrastructure/pull/1058#discussion_r1315856393

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/ET-4985)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: b2b_personalization/tests/api/snippet_size.hurl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+POST {{bo_host}}/documents
+authorizationToken: {{bo_token}}
+X-Tenant-Id: {{tenant_id}}
+Content-Type: application/json
+
+{
+    "documents": [
+        { "id": "0001",
+            "snippet": "{{snippet_string}}."
```
Comment (rustonaut) [inline b2b_personalization/tests/api/snippet_size.hurl:43]: :question:  isn't that the exact same string which was used for the ok test above?

**Review by rustonaut — COMMENTED** [review]
:question:  I don't quite understand why we add this tests there as they don't seem to test deployment aspects but things which are (or if not should be) tested by the integration test suite

especially given that they have no way to reliable cleanup state or wait for the ingestion to be fully propagated to the search API they are rather brittle test as far as I can tell

File: b2b_personalization/tests/api/cors.hurl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# [Asserts]
+# header "access-control-allow-methods" contains "GET"
+# header "access-control-allow-methods" contains "POST"
+# header "access-control-allow-methods" contains "PUT"
+# header "access-control-allow-methods" contains "PATCH"
+# header "access-control-allow-methods" contains "DELETE"
+# header "access-control-allow-methods" contains "OPTIONS"
+# header "access-control-allow-methods" contains "CONNECT"
+# header "access-control-allow-methods" contains "HEAD"
+# header "access-control-allow-methods" contains "TRACE"
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply b2b_personalization/tests/api/cors.hurl:42]: The test is currently failing. I created a task to investigate why it is failing https://xainag.atlassian.net/browse/ET-4724. Afaik we wanted to allow CORS also on the back office.

File: b2b_personalization/tests/api/properties_size.hurl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+DELETE {{bo_host}}/documents
+authorizationToken: {{bo_token}}
+X-Tenant-Id: {{tenant_id}}
+Content-Type: application/json
+
+{
+  "documents": [
+    "0001"
+  ]
+}
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply b2b_personalization/tests/api/properties_size.hurl:23]: it's more a clean up. the test before uploads a document and after that we delete it in order to keep the data base clean.

File: b2b_personalization/tests/api/properties_string_size.hurl [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+authorizationToken: {{bo_token}}
+X-Tenant-Id: {{tenant_id}}
+Content-Type: application/json
+
+{
+    "documents": [
+        { "id": "0001",
+            "snippet": "snippet",
+            "properties" : {
+                "text": "{{properties_string_string}}."
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply b2b_personalization/tests/api/properties_string_size.hurl:48]: the only difference is the `.` at the end. If you prefer, we can also use a different char to make it more.

[the remark this answers is not in the document]
Comment (rustonaut) [reply b2b_personalization/tests/api/properties_string_size.hurl:48]: yes that would be nice the trailing `.` is really easy to overlook

**Review by rustonaut — APPROVED** [review]
lgtm

changing that trailing singe `.` to idk. `...`  or similar would still be nice, trailing dots like that are really easy to overlook

**Review by rustonaut — APPROVED** [review]
lgtm

**Pull request merged**